api/desecapi/views.py

```diff
@@ -70,14 +70,19 @@ class DomainViewMixin:
     def get_serializer_context(self):
         return {**super().get_serializer_context(), 'domain': self.domain}
 
-    def initial(self, request, *args, **kwargs):
+    @property
+    def domain(self):
         # noinspection PyUnresolvedReferences
-        super().initial(request, *args, **kwargs)
+        return self.request.user.domains.get(name=self.kwargs['name'])
+
+    def check_throttles(self, request):
         try:
-            # noinspection PyAttributeOutsideInit, PyUnresolvedReferences
-            self.domain = self.request.user.domains.get(name=self.kwargs['name'])
+            # noinspection PyStatementEffect
+            self.domain  # call the method behind the property
```

Comment:
that's a weird usage of a property. Maybe convert it to a method?

Reply:
right. Did it differently now